pkg/hubble/parser/seven/http.go

```diff
@@ -80,3 +84,36 @@ func (p *Parser) httpSummary(flowType accesslog.FlowType, http *accesslog.LogRec
 	}
 	return ""
 }
+
+// filterHeader receives a key-value pair of an http header along with an HttpHeadersList.
+// Based on the allow/deny lists of the provided HttpHeadersList it returns the original value
+// or the redacted constant "HUBBLE_REDACTED" accordingly:
+//  1. If HttpHeadersList is not nil (meaning that hubble.redact feature is enabled) but both allow/deny lists are empty then the value of the
+//     header is redacted by default.
+//  2. If the header's key is contained in the allow list then the value
+//     of the header will not be redacted.
+//  3. If the header's key is contained in the deny list then the value
+//     of the header will be redacted.
+//  4. If none of the above happens, then if there is any allow list defined then the value will be redacted
+//     otherwise if there is a deny list defined the value will not be redacted.
+func filterHeader(key string, value string, headersList *options.HttpHeadersList) string {
+	if headersList != nil {
+		if len(headersList.Allow) == 0 && len(headersList.Deny) == 0 {
```

Comment:
The way this is written feels somewhat error prone. I worry that a future refactor will result in all headers being redacted by accident. Perhaps `options.HttpHeadersList` should be `options.HubbleRedactSettings` with a boolean in there for when `hubble.redact` is enabled, so that we do not need to rely on nulls and this somewhat implicit logic.

Reply:
Hey, this comes directly from https://github.com/cilium/cilium/blob/96265caefa91a788ec7d1407de8b2607b091ec5e/pkg/hubble/parser/options/options.go#L20. Do you suggest that we should replace this with a new field called `options.HubbleRedactSettings`? Since it's not quite clear how this should look like could you elaborate more with an example?

Also keep in mind that `filterHeader` function only depends on the http headers allow/deny lists and hence that's why a generic `hubble.redact` boolean might be a bit redundant.

In any case I can ofc change it accordingly if we clarify what exactly this should like.

Reply:
> Do you suggest that we should replace this with a new field called options.HubbleRedactSettings

Yes, a new struct, something like:
```
type HubbleRedactSettings struct {
  Enabled bool
  Headers *HttpHeadersList
}
```

> Also keep in mind that filterHeader function only depends on the http headers allow/deny lists and hence that's why a generic hubble.redact boolean might be a bit redundant.

Well, it also relies on the `*options.HttpHeadersList`, whether or not it's nil makes a difference. While pointers _can_ be used to enable/disable the feature, it feels like a foot-gun in this case, which is why I'm suggesting adding a new type with a boolean field to distinguish between "default redaction behavior" and "explicit redaction via allowed/denied lists".


What do you think?

Reply:
I see! Thanks for the additional explanation. I tried to cover this with [842a01a](https://github.com/cilium/cilium/pull/26724/commits/842a01aa97d28ef5dc003bb657d2ef13cabcf1d4). Let me know if that's what you had in mind :).